src/chain/chain.rs

```diff
@@ -238,6 +238,7 @@ impl Chain {
                 Some((event, proofs)) => (event, proofs),
             }
         };
+        self.add_expectation(event.clone(), &proofs);
```

Comment:
Why not directly into `poll_event` ?

Reply:
don't want to change the current API (as we need the `proofs` info for the voters)

Reply:
Why do you not want to change the current function?

Reply:
The tracker is an extra feature being appended to chain_accumulator.
The existing tests uses the `poll_event` function a lot, and having an extra `proof` required (which is not related to thos tests) in the function interface causes a hassle to update and readers to follow what the test is really for.

Reply:
Sorry, I do not thing the test update would be very big. essentially the additional parameter can be part of TestData, and because we are only using the ChainAccumulator in the test for a single result, we do not expect any unresponsive in any of the tests. One or more additional test will be needed.

However, I do not think existing code should be driving such a simple interface.
The reason it make sense for me together is that it clarify the invariant that the events we are looking for in unresponsive are the one that reached consensus last. You can still somewhat do that with and additional function, but there would be a need for clear benefit, I am not seeing it yet.

Reply:
Probably, after we get the final decision that this the right place (chain_accumulator) to track, I will merge this `add_expection` into the `poll_event`.